diff --git "a/rlImGui.cpp" "b/rlImGui.cpp"
index b5bde00..e2cb654 100644
--- "a/rlImGui.cpp"
+++ "b/rlImGui.cpp"
@@ -1,3 +1,5 @@
+// clang-format off
+
 /**********************************************************************************************
 *
 *   raylibExtras * Utilities and Shared Components for Raylib
@@ -91,11 +93,24 @@ void ImGui_ImplRaylib_FreeBackendData()
     MemFree(ImGui::GetPlatformIO().Renderer_RenderState);
 }
 
+#if defined(PLATFORM_ANDROID)
+#include <sys/system_properties.h>
 
-Vector2 GetDisplayScale()
-{
+inline float get_android_scale() {
+    char prop_value[PROP_VALUE_MAX];
+    if (__system_property_get("ro.sf.lcd_density", prop_value) > 0) {
+        float dpi = std::atof(prop_value);
+        return dpi / 160.0f;
+    }
+    return 1.0f;
+}
+#endif
+
+Vector2 GetDisplayScale() {
 #if defined(__EMSCRIPTEN__)
-    return Vector2{ 1,1 };
+    return Vector2{1, 1};
+#elif defined(PLATFORM_ANDROID)
+    return Vector2{get_android_scale(), get_android_scale()};
 #else
     return GetWindowScaleDPI();
 #endif
@@ -125,7 +140,17 @@ static void ImGuiNewFrame(float deltaTime)
 
     Vector2 resolutionScale = GetDisplayScale();
 
-#ifndef PLATFORM_DRM
+#if defined(PLATFORM_ANDROID)
+
+    io.DisplaySize.x = float(GetScreenWidth());
+    io.DisplaySize.y = float(GetScreenHeight());
+
+#elif defined(PLATFORM_DRM)
+
+    io.DisplaySize.x = float(GetScreenWidth());
+    io.DisplaySize.y = float(GetScreenHeight());
+
+#else
     if (IsWindowFullscreen())
     {
         int monitor = GetCurrentMonitor();
@@ -138,13 +163,10 @@ static void ImGuiNewFrame(float deltaTime)
         io.DisplaySize.y = float(GetScreenHeight());
     }
 
-#if !defined(__APPLE__)
-    if (!IsWindowState(FLAG_WINDOW_HIGHDPI))
-        resolutionScale = Vector2{ 1,1 };
-#endif
-#else
-    io.DisplaySize.x = float(GetScreenWidth());
-    io.DisplaySize.y = float(GetScreenHeight());
+    #if !defined(__APPLE__)
+        if (!IsWindowState(FLAG_WINDOW_HIGHDPI))
+            resolutionScale = Vector2{ 1.0f, 1.0f };
+    #endif
 #endif
 
     io.DisplayFramebufferScale = ImVec2(resolutionScale.x, resolutionScale.y);
@@ -937,3 +959,4 @@ bool ImGui_ImplRaylib_ProcessEvents(void)
 
     return true;
 }
+// clang-format on
